internal/repository/repository.go

```diff
@@ -699,33 +693,6 @@ func (r *Repository) createIndexFromPacks(ctx context.Context, packsize map[rest
 	return invalid, nil
 }
 
-// prepareCache initializes the local cache. indexIDs is the list of IDs of
-// index files still present in the repo.
-func (r *Repository) prepareCache() error {
-	if r.Cache == nil {
-		return nil
-	}
-
-	indexIDs := r.idx.IDs()
-	debug.Log("prepare cache with %d index files", len(indexIDs))
-
-	// clear old index files
-	err := r.Cache.Clear(restic.IndexFile, indexIDs)
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "error clearing index files in cache: %v\n", err)
-	}
```

Comment:
why did you remove the whole `prepareCache` function? All pack files are only listed by `check` or `prune`, so we can't remove the step that uses the index to determine which pack files should still be kept. Please revert those changes and _only_ remove the code between line 709 and 716.

Reply:
I have reverted this change. I have removed *only* lines between 709 and 716.